### ISSUE TITLE
Installation pain points

### ISSUE THREAD

**Issue by GuiguiSurLinux** [GetValkyrie/valkyrie#24]
Just creating this for documentation purposes:

I installed vagrant-triggers through vagrant plugin, which uses ruby gems. (I had installed Ruby through aptitude.) On debian, vagrant-triggers indirectly depends on ruby2.*-dev package, since its dependency, nokogiri, does. As far as I can tell, the important part of the dev package is the mkmf library. Failing to install nokogiri at first seems to cause some problems with vagrant-triggers where you have to uninstall and reinstall it, _even after the dependency is resolved_. Although it's not our role to document the install process of the dependencies, it might be good to consider that we depend on ruby2.*-dev on Debian.

Also, we assume the user has an RSA key, otherwise we'll get this error: "File upload source file /home/guigui/.ssh/id_rsa.pub must exist". Generating an RSA key is enough to make this go away, but I prefer to use DSA. It should probably just ask for the name of the key if it doesn't find id_rsa, or maybe generate one on the fly.

**Comment by ergonlogic** [comment]
We're re-implementing the various tasks we use vagrant-triggers for, as faux plugins. That is, they use the Vagrant API, but we include them directly rather than being stand-alone gems. Look in the vagrant/lib/ dir for them. They're included as git submodules, to make re-use and isolated testing of their functionality easier. The only remaining use of vagrant-triggers is to manage vagrant-dns, which is only needed on OSX. This dependency will be removed shortly, which will mean no external Vagrant plugins will be required on Linux.

As for the SSH keys, it shouldn't really matter too much, since we basically just copy it into `/var/aegir/.ssh/authorized_keys`. It should be pretty easy to just fallback to `~/id_dsa.pub` if `~/id_rsa.pub` doesn't exist.

**Comment by ergonlogic** [comment]
I split this into #42 and #43.
